migration/migrator.go

```diff
@@ -325,13 +325,19 @@ func (m *StateMigrator) FinalizeTransition(transitionBlock types.Block) {
 }
 
 func (m *StateMigrator) waitForMigrationReady(target *types.Header) {
+	safe := m.backend.BlockChain().CurrentSafeBlock()
+	// There is no gap between the safe block and the unsafe block, no waiting is required.
+	if safe != nil && safe.Hash() == target.Hash() {
```

Comment:
How about checking the block number? It seems little bit awkward to compare the block hash to check if it's the same height